### PR TITLE
Milestone 01: Morse Decoder

### PR DESCRIPTION
**Pull request by mavericks-db**
## Morse Decoder

**This branch includes:**

- [x] Create a method to decode a Morse code character, takes a string parameter, and return the corresponding character in uppercase (e.g. decode_char(".-") returns "A").
- [x] Create a method to decode an entire word in Morse code, takes a string parameter, and return the string representation. Every character in a word will be separated by a single space (e.g. decode_word("-- -.--") returns "MY").
- [x] Create a method to decode the entire message in Morse code, takes a string parameter, and return the string representation. Every word will be separated by 3 spaces (e.g. decode("-- -.--   -. .- -- .") returns "MY NAME").
- [x] Reuse methods whenever possible. Make sure that your code is DRY.
- [x] Once your code is ready use it to decode the message from the old bottle:
.-   -... --- -..-   ..-. ..- .-.. .-..   --- ..-.   .-. ..- -... .. . ...

International_Morse_Code svg

### CODE REVIEW

File: decoder.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+puts decode_char(x)
+
+# split sentence in worlds
+def split_world(phrase)
+  if phrase.is_a? String
+    phrase.split('   ')
+  else
+    put 'Your inpu should be a string in double or single quote !'
+  end
+end
```
Comment (Mre55) [inline decoder.rb:61]: Kindly, implement the proper split word method which splits input word by single space and it reuses the `#decode_char` method to translate a single char symbol into a letter,  it combines translated chars into a single word to be returned as the final output `p decode_word('-- -.--')` or `decode_word('-- -.--')` should return `"MY"`. And fix the typo of the method name it should be `split_word` in your case and fix the typo in line 59 too.

image

**Review by Mre55 — CHANGES_REQUESTED** [review]
**Hi @mavericks-db and @Baroka-wp,**

You have done a great job so far! 💪 👏 
You still need to work on a few issues before moving on to the next project, but you're almost there!

## To highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The README is well documented ✅ 

## Required Changes ♻️🛠️ 
_Check the comments under the review._




## Optional suggestions 🟡 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
# Status: Approved :tada: :tada: 

Hi @mavericks-db , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mavericks-db** [comment]
Thank you 👍

**Pull request merged**